src/net.zig

```diff
@@ -0,0 +1,746 @@
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn init(host: [:0]const u8) Error!Address {
+        return .{
+            .value = try errors.wrapCallNull(*c.NET_Address, c.NET_ResolveHostname(host.ptr)),
+        };
+    }
+
+    /// Drop a reference to an `net.Address`.
+    ///
+    /// ## Remarks
+    /// Since several pieces of the library might share a single `net.Address`, including a background thread that's working on resolving, these objects are referenced counted.
+    /// This allows everything that's using it to declare they still want it, and drop their reference to the address when they are done with it.
+    /// The object's resources are freed when the last reference is dropped.
+    /// This function drops a reference to an `net.Address`, decreasing its reference count by one.
+    /// The documentation will tell you when the app has to explicitly unref an address.
+    /// For example, `net.Address.init()` creates addresses that are already referenced, so the caller needs to call `deinit()` when done.
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn deinit(self: Address) void {
+        c.NET_UnrefAddress(self.value);
+    }
+
+    /// Add a reference to an `net.Address`.
+    ///
+    /// ## Return Value
+    /// Returns the same address that was passed as a parameter.
+    ///
+    /// ## Remarks
+    /// Since several pieces of the library might share a single `net.Address`, these objects are referenced counted.
+    /// This function adds a reference to an `net.Address`, increasing its reference count by one.
+    /// Generally you only have to explicit ref an address when you have different parts of your own app that will be sharing an address.
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn ref(self: Address) Address {
+        return .{ .value = c.NET_RefAddress(self.value) };
+    }
+
+    /// Block until an address is resolved.
+    ///
+    /// ## Function Parameters
+    /// * `self`: The `net.Address` object to wait on.
+    /// * `timeout`: Number of milliseconds to wait for resolution to complete.
+    ///   -1 to wait indefinitely, 0 to check once without waiting.
+    ///
+    /// ## Return Value
+    /// Returns `true` if successfully resolved, `false` if the timeout was reached.
+    ///
+    /// ## Thread Safety
+    /// It is safe to call this function from any thread, and several threads can block on the same address simultaneously.
+    ///
+    /// ## Version
+    /// This function is available since SDL_net 3.0.0.
+    pub fn waitUntilResolved(self: Address, timeout: i32) Error!bool {
```

Comment:
For these functions in general, I typically put a comma after the last argument so each argument is its own line (even for single argument functions).